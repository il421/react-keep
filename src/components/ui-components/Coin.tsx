import React, { memo } from "react";
import ReactTooltip from "react-tooltip";
import "../../styles/ui-components/_coin.scss";
import { getRandomColor, getShortText, JustifyContent } from "../../common";
import { FlexBox } from "./FlexBox";

export interface CoinBaseProps {
  email: string | null;
  name: string | null;
  url?: string | null;
  showTooltip?: boolean;
  className?: string;
}
const CoinBase: React.FunctionComponent<CoinBaseProps> = ({
  email,
  name,
  url,
  showTooltip,
  className,
}) => {
  return (
    <div data-tip={name ?? email}>
      <FlexBox
        justifyContent={JustifyContent.center}
        className={`coin ${className}`}
      >
        {url ? (
          <img
            src={url}
            alt={`Photo of ${name ?? email}`}
            width={25}
            height={25}
          />
        ) : (
          <span style={{ backgroundColor: getRandomColor() }}>
            {getShortText(name ?? email!, 2, false)}
          </span>
        )}
      </FlexBox>
      {showTooltip && (
        <ReactTooltip
          place="top"
          type="dark"
          effect="solid"
          className="coin__tooltip"
        />
      )}
    </div>
  );
};

export const Coin = memo(CoinBase);
