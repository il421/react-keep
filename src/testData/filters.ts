import { Filters } from "../store/store.types";

export const filter = {
  tag: "YpjypoYGueKBUukRPQ5F",
  search: "Important",
};

export const filters: Filters = {
  tagFilters: ["1"],
  search: "Important",
};
