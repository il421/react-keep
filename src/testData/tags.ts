import { Tag } from "../store/store.types";

export const tags: Tag[] = [
  {
    id: "1",
    value: "Tag1",
  },
  {
    id: "2",
    value: "Tag2",
  },
];
