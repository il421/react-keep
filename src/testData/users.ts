export const user = {
  email: "app.ilya.testing@gmail.com",
  password: "{SECURE_TEST_USER_PASS}2",
  uid: "qIxSzOgxdGcEFgG9JZg9LyWVLH2345678901",
  firstName: "Ilya",
  lastName: "Suglobov",
  url: null,
};
