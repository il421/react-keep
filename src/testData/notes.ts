import { Note } from "../store/store.types";
import { NoteType } from "../components/notes";
import { tags } from "./tags";

export const notes: Note[] = [
  {
    type: NoteType.text,
    id: "1",
    title: "1",
    content: "text",
    tags: [tags[0].id],
    color: "#1",
    important: false,
    archive: false,
    createdAt: 100000,
    updatedAt: 100005,
  },
  {
    type: NoteType.list,
    id: "2",
    title: "2",
    content: [
      {
        id: "2-1",
        content: "dog",
        checked: false,
      },
      {
        id: "2-2",
        content: "cat",
        checked: true,
      },
    ],
    tags: [tags[0].id, tags[1].id],
    color: "#2",
    important: true,
    archive: false,
    createdAt: 100000,
    updatedAt: 100005,
  },
  {
    type: NoteType.image,
    id: "3",
    title: "3",
    content: { text: "text", imageUrl: "imageUrl" },
    tags: [tags[1].id],
    color: "#3",
    important: false,
    archive: true,
    createdAt: 100000,
    updatedAt: 100005,
  },
];

export const newNote: Omit<Note, "id" | "createdAt" | "updatedAt"> = {
  color: "#cbf0f8",
  important: false,
  archive: false,
  tags: [tags[0].id],
  type: NoteType.text,
  content: "Dont forget to make a call",
  title: "Call",
};

export const updatedNote: Omit<Note, "id"> = {
  color: "#cbf0f8",
  important: false,
  archive: false,
  tags: [tags[0].id],
  type: NoteType.text,
  content: "Dont forget to make a call",
  title: "Call",
  createdAt: 54353534,
  updatedAt: 43432432,
};
