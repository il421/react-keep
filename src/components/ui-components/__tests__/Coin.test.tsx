import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { Coin, CoinBaseProps } from "../Coin";
import toJson from "enzyme-to-json";
import { user } from "../../../testData/users";

let props: CoinBaseProps, wrapper: ShallowWrapper;

beforeEach(() => {
  props = {
    url: "url",
    name: user.firstName,
    email: user.email,
  };
});

test("should render Coin", () => {
  wrapper = shallow(<Coin {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
