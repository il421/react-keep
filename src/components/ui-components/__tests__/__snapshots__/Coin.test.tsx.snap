// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Coin 1`] = `
<div
  data-tip="Ilya"
>
  <FlexBox
    className="coin undefined"
    justifyContent="center"
  >
    <img
      alt="Photo of Ilya"
      height={25}
      src="url"
      width={25}
    />
  </FlexBox>
</div>
`;
